src/master-playlist-controller.js

```diff
@@ -615,6 +619,9 @@ export class MasterPlaylistController extends videojs.EventTarget {
         // if we shouldn't switch to the next playlist, do nothing
         if (!this.shouldSwitchToMedia_(nextPlaylist)) {
           this.logger_(`earlyabort triggered, but we will not be switching from ${currentPlaylist.id} -> ${nextPlaylist.id}.`);
+          this.mainSegmentLoader_.earlyabortTimer = window.setTimeout(() => {
+            this.mainSegmentLoader_.trigger('earlyabort');
+          }, 10 * 1000);
```

Comment:
10s might be too long, what do you think about 5s?

Reply:
Probably should be target duration dependent or something, maybe even forward buffer? Made it 10s just for starters.